hs_core/hydroshare/utils.py

```diff
@@ -251,10 +249,7 @@ def get_file_from_irods(res_file):
     :return: location of the copied file
     """
     res = res_file.resource
-    if res_file.fed_resource_file or res_file.fed_resource_file_name_or_path:
-        istorage = IrodsStorage('federated')
-    else:
-        istorage = IrodsStorage()
+    istorage = res.get_irods_storage()
```

Comment:
@pkdash I believe you will have to revert back to the original code since ```get_irods_storage``` does not account for the specific conditions needed for checking whether to use federated IrodsStorage or not. This routine is used to retrieve a file from a federated zone for metadata extraction before resource_federation_path property is set on the resource yet, so you cannot change the code to use ```get_irods_storage``` for this special case.